apps/test-suite/tests/SMS.ios.js

```diff
@@ -14,6 +15,40 @@ export function test({ describe, it, expect }) {
           // TODO: Bacon: Build an API to close the UI Controller
           await SMS.sendSMSAsync(['0123456789', '9876543210'], 'test');
         });
+
+        it(`opens an SMS composer with image`, async () => {
+          const localUri = FS.documentDirectory + 'image.png';
+          const assertExists = async expectedToExist => {
+            const { exists } = await FS.getInfoAsync(localUri);
+            if (expectedToExist) {
+              expect(exists).toBeTruthy();
+            } else {
+              expect(exists).not.toBeTruthy();
+            }
+          };
+          await FS.deleteAsync(localUri, { idempotent: true });
+          await assertExists(false);
+          const {
+            md5,
+            headers,
+          } = await FS.downloadAsync(
+            'https://s3-us-west-1.amazonaws.com/test-suite-data/avatar2.png',
+            localUri,
+            { md5: true }
+          );
+          expect(md5).toBe('1e02045c10b8f1145edc7c8375998f87');
+          await assertExists(true);
+          expect(headers['Content-Type']).toBe('image/png');
+          const contentUri = await FS.getContentUriAsync(localUri);
+          await SMS.sendSMSAsync(['0123456789', '9876543210'], 'test with image', {
+            contentUri,
+            androidType: 'image/png',
+            iOSType: 'public.png',
+            iOSFilename: 'image.png',
```

Comment:
This API is pretty disjointed, perhaps we could find a way to remove all android and ios specific fields.

Reply:
I'd like to see the UTI be aliased to a mime type on iOS so it's universal, and works across SMS and MailComposer.

Reply:
I think not restricting the mimeType and throwing an error in the event it cannot be matched on the iOS side may be best. 

After your feedback / chatting about it, I tried to add more types (gif, mp3, ...) but it seems like that wouldn't be helpful and would only restrict the usage. 

`isSupportedAttachmentUTI` could be helpful but the Apple's list of mimeType to UTI is pretty exhaustive. I won't be adding this for now unless you feel strongly.